tests/scalers/aws/aws_dynamodb_streams_pod_identity/aws_dynamodb_streams_pod_identity_test.go

```diff
@@ -159,55 +158,56 @@ func TestScaler(t *testing.T) {
 	cleanupDynamoDBTable(t, dbClient)
 }
 
-func setupDynamoDBStreams(t *testing.T) (*dynamodb.DynamoDB, *dynamodbstreams.DynamoDBStreams) {
-	sess := session.Must(session.NewSession())
+func setupDynamoDBStreams(t *testing.T) (*dynamodb.Client, *dynamodbstreams.Client) {
+	var dbClient *dynamodb.Client
+	var dbStreamClient *dynamodbstreams.Client
 
-	var dbClient *dynamodb.DynamoDB
-	var dbStreamClient *dynamodbstreams.DynamoDBStreams
+	configOptions := make([]func(*config.LoadOptions) error, 0)
+	configOptions = append(configOptions, config.WithRegion(awsRegion))
+	cfg, _ := config.LoadDefaultConfig(context.TODO(), configOptions...)
+	cfg.Credentials = credentials.NewStaticCredentialsProvider(awsAccessKey, awsSecretKey, "")
 
-	creds := credentials.NewStaticCredentials(awsAccessKey, awsSecretKey, "")
-	cfg := aws.NewConfig().WithRegion(awsRegion).WithCredentials(creds)
-	dbClient = dynamodb.New(sess, cfg)
-	dbStreamClient = dynamodbstreams.New(sess, cfg)
+	dbClient = dynamodb.NewFromConfig(cfg)
+	dbStreamClient = dynamodbstreams.NewFromConfig(cfg)
 
 	err := createTable(dbClient)
 	assert.NoErrorf(t, err, "cannot create dynamodb table - %s", err)
 
 	return dbClient, dbStreamClient
 }
 
-func createTable(db dynamodbiface.DynamoDBAPI) error {
-	keySchema := []*dynamodb.KeySchemaElement{
+func createTable(db *dynamodb.Client) error {
+	keySchema := []dynamodbTypes.KeySchemaElement{
 		{
 			AttributeName: aws.String("id"),
-			KeyType:       aws.String("HASH"),
+			KeyType:       dynamodbTypes.KeyTypeRange,
```

Comment:
This e2e is failing during the table creation awith the message: Invalid KeySchema: The first KeySchemaElement is not a HASH key type
I have compared this test with the not podIdentity test and they KeyType is different, I guess that's the reason:
```suggestion
			KeyType:       dynamodbTypes.KeyTypeHash,
```